### PR TITLE
[WIP] Publish Trust Flow

### PR DESCRIPTION
**Pull request by mattberridge**

### CODE REVIEW

File: src/actions/SupportActions.js [hunk 1]
```diff
@@ -1,45 +1,68 @@
...
-
-import {
-  showFlashMessage
-} from './FlashActions';
-
-import {
-  showModalWindow
-} from './ModalActions';
+export function supportPetition (trustData, dispatch) {
+  console.log('support', trustData);
```
Comment (eschaefer) [inline src/actions/SupportActions.js]: Could remove this `console.log`

File: src/components/TrustConfirmation/index.js [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
 const TrustConfirmation = ({
   isLoggedIn,
-  me
+  me,
+  action
 }) => (
   <FormLayout
     title={settings.trustConfirmationPage.title}
     intro={settings.trustConfirmationPage.intro}>
     <p>Thanks {me.firstname}, we sent an SMS verification code to <b>{me.mobile}</b></p>
```
Comment (eschaefer) [inline src/components/TrustConfirmation/index.js]: Save string in `settings`?

File: src/containers/PreviewPetition.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import PreviewPetition from 'components/PreviewPetition';
+import getPetitionPath from 'selectors/petitionPath';
+import getPetitionForm from 'selectors/petitionForm';
+// import petitionPublished from 'selectors/petitionPublished';
+import trustSubmitted from 'helpers/trustSubmitted';
+
+const PreviewPetitionContainer = withRouter(React.createClass({
+  componentWillMount () {
+    // const {
+    //   fetchPetition,
```
Comment (eschaefer) [inline src/containers/PreviewPetition.js]: Delete commented code block?

File: src/containers/PreviewPetition.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    if (trustSubmitted) {
+      // Go to petition path, or confirmation
+      router.push(isTrustedUser
+        ? getPetitionPath(petition.id)
+        : `/trust/publish/${petition.id}/confirm`
+      );
+    }
+
+    // if (nextProps.petition.hasPublished) {
+    //   this.props.router.push(`${getPetitionPath(nextProps.petition)}/published`);
```
Comment (eschaefer) [inline src/containers/PreviewPetition.js]: Delete commented code?

File: src/containers/TrustPublishConfirmation.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+const TrustPublishConfirmationContainer = withRouter(React.createClass({
+  componentWillMount () {
+    const { router, petition } = this.props;
+
+    if (petition.published) {
+      router.push(getPetitionPath(petition));
+    }
+
+    // const { router, petition, hasValidUserData, showFlashMessage } = this.props;
+    // if (!hasValidUserData) {
```
Comment (eschaefer) [inline src/containers/TrustPublishConfirmation.js]: Delete commented code

File: src/containers/TrustSupport.js [hunk 6]
```diff
@@ -8,41 +8,46 @@ import { newTrustStep } from 'actions/TrustActions';
...
+import getPetitionForm from 'selectors/petitionForm';
 import trustSubmitted from 'helpers/trustSubmitted';
 
 const TrustContainer = withRouter(React.createClass({
   componentWillMount () {
-    this.props.newTrustStep();
-    // const { router, petition, showFlashMessage } = this.props;
+    // const { router, petition } = this.props;
+
     // If a petition is not supportable, redirect to petition page
```
Comment (eschaefer) [inline src/containers/TrustSupport.js]: Delete commented code

File: src/server/render.js [hunk 7]
```diff
@@ -6,13 +6,26 @@ import { createStore, applyMiddleware } from 'redux';
...
+ *
+ * Get data on the server by adding a `fetchData` function to a container e.g.
+ *
+ * PetitionContainer.fetchData = ({ store, params }) => {
+ *   return store.dispatch(fetchPetition(params.id));
+ * };
+ *
+ * This function is run on the server before rendering, dispatching an action
+ * to pre-fill the Redux store with data for that route.
+ */
```
Comment (eschaefer) [inline src/server/render.js]: Nice

File: src/containers/TrustSupport.js (hunk 6 above)
Comment (mattberridge) [inline src/containers/TrustSupport.js]: For testing 👍

File: src/containers/PreviewPetition.js (hunk 4 above)
Comment (mattberridge) [inline src/containers/PreviewPetition.js]: For testing 👍

File: src/containers/PreviewPetition.js (hunk 3 above)
Comment (mattberridge) [inline src/containers/PreviewPetition.js]: For testing 👍

File: src/components/TrustConfirmation/index.js (hunk 2 above)
Comment (mattberridge) [inline src/components/TrustConfirmation/index.js]: Ahh good spot. I've also got to do the button strings as well.
But i'll do this in the content / styling branch I guess.

File: src/server/render.js (hunk 7 above)
Comment (mattberridge) [inline src/server/render.js]: 😄

**Pull request merged**